tests/test_algorithm/test_configuration/test_maps/test_dense_map_3d.py

```diff
@@ -0,0 +1,319 @@
+            [[DenseMap.WALL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID], [DenseMap.AGENT_ID, DenseMap.CLEAR_ID, DenseMap.EXTENDED_WALL]],
+            [[DenseMap.WALL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID], [DenseMap.GOAL_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID]]
+        ])
+        map2: DenseMap = DenseMap([
+            [[DenseMap.WALL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID], [DenseMap.AGENT_ID, DenseMap.CLEAR_ID, DenseMap.EXTENDED_WALL]],
+            [[DenseMap.WALL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID], [DenseMap.CLEAR_ID, DenseMap.CLEAR_ID, DenseMap.GOAL_ID]]
+        ])
+        self.assertNotEqual(map1, map2)
+
+    def test_ne_wall(self) -> None:
+        map1: DenseMap = DenseMap([
+            [[DenseMap.WALL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID], [DenseMap.AGENT_ID, DenseMap.CLEAR_ID, DenseMap.EXTENDED_WALL]],
+            [[DenseMap.WALL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID], [DenseMap.GOAL_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID]]
+        ])
+        map2: DenseMap = DenseMap([
+            [[DenseMap.WALL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID], [DenseMap.AGENT_ID, DenseMap.WALL_ID, DenseMap.EXTENDED_WALL]],
+            [[DenseMap.WALL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID], [DenseMap.GOAL_ID, DenseMap.WALL_ID, DenseMap.WALL_ID]]
+        ])
+        self.assertNotEqual(map1, map2)
+
+    def test_ne_all(self) -> None:
+        map1: DenseMap = DenseMap([
+            [[DenseMap.CLEAR_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID], 
+             [DenseMap.CLEAR_ID, DenseMap.WALL_ID, DenseMap.CLEAR_ID], 
+             [DenseMap.AGENT_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID]
+            ], 
+            [[DenseMap.CLEAR_ID, DenseMap.WALL_ID, DenseMap.CLEAR_ID], 
+             [DenseMap.CLEAR_ID, DenseMap.WALL_ID, DenseMap.CLEAR_ID], 
+             [DenseMap.WALL_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID]
+            ], 
+            [[DenseMap.EXTENDED_WALL, DenseMap.CLEAR_ID, DenseMap.GOAL_ID], 
+             [DenseMap.CLEAR_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID], 
+             [DenseMap.CLEAR_ID, DenseMap.WALL_ID, DenseMap.CLEAR_ID]
+            ]
+        ])
+        map2: DenseMap = DenseMap([
+            [[DenseMap.AGENT_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID], 
+             [DenseMap.CLEAR_ID, DenseMap.EXTENDED_WALL, DenseMap.CLEAR_ID], 
+             [DenseMap.CLEAR_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID]
+            ], 
+            [[DenseMap.CLEAR_ID, DenseMap.CLEAR_ID, DenseMap.WALL_ID], 
+             [DenseMap.CLEAR_ID, DenseMap.CLEAR_ID, DenseMap.WALL_ID], 
+             [DenseMap.WALL_ID, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID]
+            ], 
+            [[DenseMap.WALL_ID, DenseMap.GOAL_ID, DenseMap.WALL_ID], 
+             [DenseMap.EXTENDED_WALL, DenseMap.CLEAR_ID, DenseMap.CLEAR_ID], 
+             [DenseMap.CLEAR_ID, DenseMap.WALL_ID, DenseMap.CLEAR_ID]
+            ]
+        ])
+        self.assertNotEqual(map1, map2)
+
+    # TODO: sort out sparse map stuff (currently 2D)
```

Comment:
@zbpgithub can we sort out these TODOs before this is merged?

Reply:
I think John is working on sparse maps so I was going to address these tests then, however I can implement failing tests for now if you'd prefer?